src/main/java/io/github/dsheirer/record/wave/WaveWriter.java

```diff
@@ -105,7 +105,7 @@ public WaveWriter(AudioFormat format, Path file, long maxSize) throws IOExceptio
      */
     public WaveWriter(AudioFormat format, Path file) throws IOException
     {
-        this(format, file, Integer.MAX_VALUE * 2);
+        this(format, file, 0);
```

Comment:
because of Java type conversion and promotion rules, multiplication of `Integer.MAX_VALUE` (of type `int`) and `2` (of type `int`) will overflow (to type `int`) even though the third argument is of type `long`.

what we likely want here is to default to `MAX_WAVE_SIZE`, so we can pass 0 or just `MAX_WAVE_SIZE` if you prefer

Reply:
thanks maybe this will fix wav recordings (a feature i use) maybe some calls getting cut off after 8 minutes or so on long (large) files